iml-manager-cli/src/ostpool.rs

```diff
@@ -94,69 +94,72 @@ async fn pool_lookup(fsname: &str, poolname: &str) -> Result<OstPoolApi, ImlMana
     Ok(pool)
 }
 
-pub async fn ostpool_cli(command: OstPoolCommand) -> Result<(), ImlManagerCliError> {
-    let term = Term::stdout();
-    match command {
-        OstPoolCommand::List { fsname } => {
-            let xs: Vec<_> = match fsname {
-                Some(fsname) => {
-                    let fs: Filesystem =
-                        wrap_fut("Fetching Filesystem ...", get_one(vec![("name", &fsname)]))
-                            .await?;
-                    let pools: ApiList<OstPoolApi> = wrap_fut(
-                        "Fetching OstPools...",
-                        get(
-                            OstPoolApi::endpoint_name(),
-                            vec![("limit", 0), ("filesystem", fs.id)],
-                        ),
-                    )
-                    .await?;
-                    pools
-                        .objects
-                        .into_iter()
-                        .map(|p| vec![fsname.clone(), p.ost.name, p.ost.osts.len().to_string()])
-                        .collect()
-                }
-                None => {
-                    let pools: ApiList<OstPoolApi> =
-                        wrap_fut("Fetching OstPools...", get_all()).await?;
-
-                    let fs_ids = pools
-                        .objects
-                        .iter()
-                        .filter_map(|p| {
-                            let id = extract_api_id(&p.ost.filesystem);
+async fn ostpool_list(fsname: Option<String>) -> Result<(), ImlManagerCliError> {
+    let xs: Vec<_> = match fsname {
+        Some(fsname) => {
+            let fs: Filesystem =
+                wrap_fut("Fetching Filesystem ...", get_one(vec![("name", &fsname)])).await?;
+            let pools: ApiList<OstPoolApi> = wrap_fut(
+                "Fetching OstPools...",
+                get(
+                    OstPoolApi::endpoint_name(),
+                    vec![("limit", 0), ("filesystem", fs.id)],
+                ),
+            )
+            .await?;
+            pools
+                .objects
+                .into_iter()
+                .map(|p| vec![fsname.clone(), p.ost.name, p.ost.osts.len().to_string()])
+                .collect()
+        }
+        None => {
+            let pools: ApiList<OstPoolApi> = wrap_fut("Fetching OstPools...", get_all()).await?;
 
-                            id.map(|x| x.to_string())
-                        })
-                        .collect::<std::collections::HashSet<String>>();
+            let fs_ids = pools
+                .objects
+                .iter()
+                .filter_map(|p| {
+                    let id = extract_api_id(&p.ost.filesystem);
 
-                    let fs_ids = Vec::from_iter(fs_ids).join(",");
+                    id.map(|x| x.to_string())
+                })
+                .collect::<std::collections::HashSet<String>>();
 
-                    let mut query = Filesystem::query();
+            let fs_ids = Vec::from_iter(fs_ids).join(",");
```

Comment:
nit: I don't like shadowing :)

Maybe it could be `... ..collect::<Vec<String>>().join(",");`, but ok.